src/Uno.UI/Microsoft/UI/Xaml/Controls/NumberBox/NumberBox.cs

```diff
@@ -534,11 +534,7 @@ private void ValidateInput()
 
 					var value = AcceptsExpression
 						? NumberBoxParser.Compute(text, numberParser)
-						: ApiInformation.IsTypePresent(numberParser?.GetType().FullName)
-							? numberParser.ParseDouble(text)
-							: double.TryParse(text, out var v)
-								? (double?)v
-								: null;
+						: numberParser.ParseDouble(text);
```

Comment:
@Youssef1313 This happens because `INumberParser` is custom, which aligns with WinUI's behavior of returning false. However, since they don't perform this check, it works fine for them. I believe we no longer need this as all of our built-in formatters are now implemented.

Reply:
The behavior of IsTypePresent is still questionable. Even if WinUI is buggy, this is a case where we may want to not follow WinUI. @MartinZikmund what do you think?

**NOTE:** If all our `INumerParser`s are now implemented, I'm okay with the change for sure. But we still need to make sure we want IsTypePresent to behave the way we want it to be.

**NOTE 2:** There is another call to `IsTypePresent` in NumberBox that may need to be removed as well

Reply:
I agree it feels safe to make it return `true` for user types

Reply:
As continued in our internal discussion with @MartinZikmund, I will remove the `IsTypePresent`s checks here. And I will have a look into making an adjustment the the behavior in a separate PR.

Reply:
That makes sense. Thank you!